mirdata/billboard.py

```diff
@@ -0,0 +1,453 @@
+
+    Attributes:
+        track_id (str): track id
+        audio_path (str): audio path of the track
+        chart date: release date of the track
+        target rank: 
+        actual rank: 
+        title: title of the track
+        artist: artist name
+        peak rank: 
+        weeks on chart: 
+    """
+
+    def __init__(self, track_id, data_home=None):
+        if track_id not in DATA.index:
+            raise ValueError("{} is not a valid track ID in Salami".format(track_id))
+
+        self.track_id = track_id
+
+        if data_home is None:
+            data_home = utils.get_default_dataset_path(DATASET_DIR)
+
+        self._data_home = data_home
+        self._track_paths = DATA.index[track_id]
+
+        metadata = DATA.metadata(data_home)
+        if metadata is not None and track_id in metadata.keys():
+            self._track_metadata = metadata[track_id]
+        else:
+            # annotations with missing metadata
+            self._track_metadata = {
+                "chart_date": None,
+                "target_rank": None,
+                "actual_rank": None,
+                "title": None,
+                "artist": None,
+                "peak_rank": None,
+                "weeks_on_chart": None,
+            }
+
+        self.salami_path = os.path.join(self._data_home, self._track_paths["salami"][0])
+        self.lab_full_path = os.path.join(
+            self._data_home, self._track_paths["lab_full"][0]
+        )
+        self.lab_majmin7_path = os.path.join(
+            self._data_home, self._track_paths["lab_majmin7"][0]
+        )
+        self.lab_majmin7inv_path = os.path.join(
+            self._data_home, self._track_paths["lab_majmin7inv"][0]
+        )
+        self.lab_majmin_path = os.path.join(
+            self._data_home, self._track_paths["lab_majmin"][0]
+        )
+        self.lab_majmininv_path = os.path.join(
+            self._data_home, self._track_paths["lab_majmininv"][0]
+        )
+
+        self.audio_path = os.path.join(self._data_home, self._track_paths["audio"][0])
+        self.chart_date = self._track_metadata["chart_date"]
+        self.target_rank = self._track_metadata["target_rank"]
+        self.actual_rank = self._track_metadata["actual_rank"]
+        self.title = self._track_metadata["title"]
+        self.artist = self._track_metadata["artist"]
+        self.peak_rank = self._track_metadata["peak_rank"]
+        self.weeks_on_chart = self._track_metadata["weeks_on_chart"]
+
+    def __repr__(self):
+        repr_string = (
+            "Billboard Track(track_id={}, audio_path={}, "
+            + "chart_date={}, "
+            + "title={}, artist={}, "
+            + "SectionData('intervals', 'labels')"
+        )
+        return repr_string.format(
+            self.track_id, self.audio_path, self.chart_date, self.title, self.artist
+        )
+
+    @utils.cached_property
+    def chords(self):
```

Comment:
I like this idea but another way of doing this would be to have one chord per type, so you can plug it directly to mir_eval. @rabitt thoughts?

Reply:
Yeah I agree - a bit bulkier to write. Something like:
```python
@utils.cached_property
def chords_full(self):
    ...

@utils.cached_property
def chords_majmin7(self):
    ...

# etc
```